Fix D3D SECURITY_ATTRIBUTES struct decode

dgraves opened the pull request:
D3D struct encoding was using the custom SECURITY_ATTRIBUTES encoder defined for Vulkan, but decoding was using an incompatible generated decoder, resulting in an access violation during the decoding process. This change updates the code generator to use the custom SECURITY_ATTRIBUTES decoder, with the same logic used to process the LARGE_INTEGER struct, which also has a custom decoder, applied to the SECURITY_ATTRIBUTES struct.

A fix for a potential null dereference when decoding the SECURITY_DESCRIPTOR member of SECURITY_ATTRIBUTES, which impacts both D3D and Vulkan, is also included.

dgraves committed:
Fix D3D SECURITY_ATTRIBUTES struct decode

D3D struct encoding was using the custom SECURITY_ATTRIBUTES encoder
defined for Vulkan, but decoding was using an incompatible generated
decoder, resulting in an access violation during the decoding process.
This change updates the code generator to use the custom
SECURITY_ATTRIBUTES decoder, with the same logic used to process the
LARGE_INTEGER struct, which also has a custom decoder, applied to the
SECURITY_ATTRIBUTES struct.

dgraves committed:
Fix SECURITY_DESCRIPTOR decoding

Fix potential null dereference with SECURITY_DESCIPTOR SID member
decoding.

ci-tester-lunarg commented:
CI gfxreconstruct build queued with queue ID 466123.

bradgrantham-lunarg reviewed (changes requested):
Just the one thought about factoring out non-API-specific bits

ci-tester-lunarg commented:
CI gfxreconstruct build # 7028 running.

ci-tester-lunarg commented:
CI gfxreconstruct build # 7028 passed.

dgraves committed:
Move API agnostic struct code to common files

Moves API agnostic custom struct encode, decode, and to_json routines to
a comnon set of files that are shared by the DX12 and Vulkan specific
code.

dgraves committed:
Move LARGE_INTEGER encode/decode to common

dgraves committed:
Codegen helper for commo structs

Adds a new common folder with a class to identify common structs, which
are shared by multiple APIs.

bradgrantham-lunarg commented:
Looks pretty good and I think I have just the concern about factoring out the functionality.  Thank you @dgraves !

ci-tester-lunarg commented:
CI gfxreconstruct build queued with queue ID 520056.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LunarG/gfxreconstruct?pullRequest=2251) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LunarG/gfxreconstruct?pullRequest=2251) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LunarG/gfxreconstruct?pullRequest=2251) it.</sub>
